client/log_verifier.go

```diff
@@ -53,17 +54,20 @@ func NewLogVerifier(hasher hashers.LogHasher, pubKey crypto.PublicKey, sigHash c
 // NewLogVerifierFromTree creates a new LogVerifier using the algorithms
 // specified by *trillian.Tree.
 func NewLogVerifierFromTree(config *trillian.Tree) (*LogVerifier, error) {
+	if config == nil {
+		return nil, errors.New("client: NewLogVerifierFromTree(): nil config")
+	}
 	log, pLog := trillian.TreeType_LOG, trillian.TreeType_PREORDERED_LOG
 	if got := config.TreeType; got != log && got != pLog {
 		return nil, fmt.Errorf("client: NewLogVerifierFromTree(): TreeType: %v, want %v or %v", got, log, pLog)
 	}
 
-	logHasher, err := hashers.NewLogHasher(config.GetHashStrategy())
+	logHasher, err := hashers.NewLogHasher(config.HashStrategy)
 	if err != nil {
 		return nil, fmt.Errorf("client: NewLogVerifierFromTree(): NewLogHasher(): %v", err)
 	}
 
-	logPubKey, err := der.UnmarshalPublicKey(config.GetPublicKey().GetDer())
+	logPubKey, err := der.UnmarshalPublicKey(config.PublicKey.GetDer())
```

Comment:
Looks like that will crash if public key is nil.

Reply:
`GetDer` will check if PublicKey is nil and won't crash.

Reply:
Is that case tested?

Reply:
Added a test